core/auditlog/auditlog.go

```diff
@@ -0,0 +1,240 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+package auditlog
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/juju/errors"
+	"github.com/juju/loggo"
+	"github.com/juju/utils"
+	"gopkg.in/natefinch/lumberjack.v2"
+)
+
+var logger = loggo.GetLogger("core.auditlog")
+
+// Call represents a high-level juju command from the juju client (or
+// other client). There'll be one Call per API connection, with zero
+// or more associated FacadeMethods.
+type Call struct {
+	Who          string `json:"who"`        // username@idm
+	What         string `json:"what"`       // "juju deploy ./foo/bar"
+	When         string `json:"when"`       // ISO 8601 to second precision
+	ModelName    string `json:"model-name"` // full representation "user/name"
+	ModelUUID    string `json:"model-uuid"`
+	CallID       string `json:"call-id"`       // uint64 in hex
+	ConnectionID string `json:"connection-id"` // uint64 in hex (using %X to match the value in log files)
+}
+
+// CallArgs is the information needed to create a method recorder.
+type CallArgs struct {
+	Who          string
+	What         string
+	When         time.Time
+	ModelName    string
+	ModelUUID    string
+	ConnectionID uint64
+}
+
+// FacadeRequest represents a call to an API facade made as part of
+// executing a specific high-level command.
+type FacadeRequest struct {
+	CallID       string `json:"call-id"`
+	ConnectionID string `json:"connection-id"`
+	RequestID    uint64 `json:"request-id"`
+	Facade       string `json:"facade"`
+	Method       string `json:"method"`
+	Version      int    `json:"version"`
+	Args         string `json:"args,omitempty"`
+}
+
+// RequestArgs is the information about an API call that we want to
+// record.
+type RequestArgs struct {
+	Facade    string
+	Method    string
+	Version   int
+	Args      string
+	RequestID uint64
+}
+
+// FacadeResponse captures any errors coming back from the API in
+// response to a request.
+type FacadeResponse struct {
```

Comment:
To make it clear this does not include any payload data (just errors), should it be called FacadeResponseErrors

Reply:
Ok